byway/src/main/java/com/google/sps/servlets/EntityServlet.java

```diff
@@ -0,0 +1,115 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Trip;
+import com.google.sps.data.UserInfo;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/entity")
+public class EntityServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    com.google.appengine.api.users.User user = userService.getCurrentUser();
+    UserInfo userInfo = addUserEntity(user);
+    List<String> userTripIds = userInfo.getTripIds();
+    ArrayList<Trip> userTrips = new ArrayList<Trip>();
+    for (String tripId : userTripIds) {
+      try {
+        Entity tripEntity = datastore.get(KeyFactory.stringToKey(tripId));
+        Trip trip = Trip.fromEntity(tripEntity);
+        userTrips.add(trip);
+      } catch (EntityNotFoundException exception) {
+        return;
+      }
+    }
+    String json = gson.toJson(userTrips);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Key tripKey = createTripEntity();
+    String keyString = KeyFactory.keyToString(tripKey);
+    System.out.println(keyString);
```

Comment:
Prefer to use `FluentLogger` for info and error messages.
https://google.github.io/flogger/

Reply:
done